File: CS/RangeDate/MyGridControl/MyGridControl.cs
using System;
using System.Linq;
using DevExpress.XtraGrid;
using DevExpress.XtraGrid.Registrator;
using System.Collections.Generic;
using DevExpress.XtraGrid.Views.Base;


namespace DateRange {
    [System.ComponentModel.DesignerCategory("")]
    public class MyGridControl : GridControl {
        protected override void RegisterAvailableViewsCore(InfoCollection collection) {
            base.RegisterAvailableViewsCore(collection);
            collection.Add(new MyGridViewInfoRegistrator());
        }
    }

    public class MyGridViewInfoRegistrator : GridInfoRegistrator {
        public override string ViewName { get { return "MyGridView"; } }
        public override BaseView CreateView(GridControl grid) {
            return new MyGridView(grid as GridControl);
        }
    }
}

File: VB/RangeDate/MyGridControl/MyGridControl.vb
Imports DevExpress.XtraGrid
Imports DevExpress.XtraGrid.Registrator
Imports DevExpress.XtraGrid.Views.Base

Namespace DateRange

    <System.ComponentModel.DesignerCategory("")>
    Public Class MyGridControl
        Inherits GridControl

        Protected Overrides Sub RegisterAvailableViewsCore(ByVal collection As InfoCollection)
            MyBase.RegisterAvailableViewsCore(collection)
            collection.Add(New MyGridViewInfoRegistrator())
        End Sub
    End Class

    Public Class MyGridViewInfoRegistrator
        Inherits GridInfoRegistrator

        Public Overrides ReadOnly Property ViewName As String
            Get
                Return "MyGridView"
            End Get
        End Property

        Public Overrides Function CreateView(ByVal grid As GridControl) As BaseView
            Return New MyGridView(TryCast(grid, GridControl))
        End Function
    End Class
End Namespace
